nilearn/datasets.py

```diff
@@ -210,33 +210,35 @@ def _get_dataset_dir(dataset_name, data_dir=None, env_vars=[],
     4. nilearn_data in the user home folder
     """
     # We build an array of successive paths by priority
+    # The boolean indicates if it is a pre_dir: in that case, we won't add the
+    # dataset name to the path.
     paths = []
 
     # Search given environment variables
-    for env_var in env_vars:
-        env_data = os.getenv(env_var, '')
-        paths.extend(env_data.split(':'))
+    for pre_dir in pre_dirs:
+        paths.extend([(d, True) for d in pre_dir.split(':')])
 
     # Check data_dir which force storage in a specific location
     if data_dir is not None:
-        paths = data_dir.split(':')
+        paths.extend([(d, False) for d in data_dir.split(':')])
     else:
         global_data = os.getenv('NILEARN_SHARED_DATA')
         if global_data is not None:
-            paths.extend(global_data.split(':'))
+            paths.extend([(d, False) for d in global_data.split(':')])
```

Comment:
In the doc, we say that single path are handled in parameter or in environment variables. Behind the scenes, we also handle several paths separated by colons.